opentelemetry-sdk/src/metrics/internal/aggregate.rs

```diff
@@ -96,7 +91,7 @@ impl<T: Number<T>> AggregateBuilder<T> {
         let filter = self.filter.as_ref().map(Arc::clone);
         move |n, mut attrs: AttributeSet| {
             if let Some(filter) = &filter {
-                attrs.retain(filter.as_ref());
+                attrs = attrs.clone_with(filter.as_ref());
```

Comment:
This will clone every HashKeyValue that passes the filter. Unfortunately, I don't think this would be easy to avoid. Not blocking this PR, maybe something we can look in the future.

Reply:
Yeah I wasn't happy about this part of the change, but the only way precreated attribute sets really worked is by making them immutable.